File: search/linear_search.c
#include<stdio.h>
/*
 Linear search
 Time complexity = O(n)
 */  
int main()
{
	int arr[] = {10, 20, 80, 30, 60, 50,  110, 100, 130, 170};
	int x;
	printf("Enter number to search:");
	scanf("%d", &x);
	for(int i=0;i<10;++i)
	{
		if(arr[i] == x)
		{
			printf("Found %d as arr[%d]\n", x, i);
			return 0;
		}
	}
	printf("Array doesn't contain your element\n");
	return 1;
}

File: search/linear_search.go
package main

import "fmt"

func search(what int, array []int) bool {
	for _, value := range array {
		if what == value {
			return true
		}
	}
	return false
}

func main() {
	array := [...]int{6, 1, 3, 3, 7, 10, 4, 4, 6, 3, 122, 381, 615, 480, 221, 98, 563, 28, 626, 439, 9, 7, 10, 10, 7, 10, 3, 5, 10, 4}
	fmt.Println("Enter a number to search in program's array")
	var i int
	fmt.Scan(&i)
	if search(i, array[:]) {
		fmt.Println("Element in array")
	} else {
		fmt.Println("Element not in array")
	}
}
